### PR TITLE
[PEPPER-1335] Implement MedicalRecordInitService

### PR DESCRIPTION
**Pull request by denniscunningham**
## Context
PEPPER-1335

The primary purpose of this PR is to implement a service to create a MedicalRecord bundle (which includes MedicalRecord, Institution, OncHistory and ParticipantRecord) via an AdminOperation. This was needed to populate synthesized OS1 participants for testing, but is generally useful as a fixup should anything go wrong with the way DSM creates MedicalRecord bundles based on an interaction with DSS.

There are several supporting  changes to enable this service (`MedicalRecordInitService`), largely because the code to do this was fragment and needed some additional tests. The PR also provoked a couple multi-file cleanups, but I will annotate those with PR comments.

The main code is in `MedicalRecordInitService` and `MedicalRecordInitServiceTest`, along with `InstitutionListRequest`. The main support for this is in `MedicalRecordService.writeInstitutionBundle`. This originated in `DDPMedicalRecordDataRequest`, and I was gradually rewriting it there, but for this PR I needed to finish the rewrite and move it. To support the rewrite of `writeInstitutionBundle`, I added `ParticipantService` (and `ParticipantServiceTest` to start centralizing the various ways DSM interacts with Participant objects. 

Other changes:

1. Expanded `ParticipantDao` to do updates, and handle a passed-in Connection.
2. Changed `UpdateLog.status` to use the `UpdateLog.UpdateStatus` enum. Switching between String and UpdateStatus was making the code harder to read, and with the current deserializer is unnecessary. Unfortunately this change rippled into several files.
3. Changed `DDPInstanceProvider` and `OsteoParticipantService` to use `DDPInstance` instead of `DDPInstanceDto`, since converting between the two meant more DB lookups.
4. Changed `OncHistoryService.createEmptyOncHistory` to detect and allow non-updates.
5. Removed `MedicalRecordUtil.updateParticipant`. It was superseded by the new ParticipantService, and did not belong in that class anyway.
6. Disabled `NewOsteoParticipantExporter`. I overlooked this prior, but it is no longer needed.
7. Added object content checkers to `InstitutionRequest` and `Institution` to support input verification in `MedicalRecordInitService`.

Changes not mentioned above will be part of my PR comments.

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have added regression test labels to the jira ticket
- [ ] I have added verification steps to the jira ticket
- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/DDPInstance.java [hunk 1]
```diff
@@ -118,6 +119,27 @@ public DDPInstance(String ddpInstanceId, String name, String baseUrl, String col
         this.displayName = displayName;
     }
 
+    @VisibleForTesting
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/DDPInstance.java:122]: This is used by a disabled test, so hopefully we can remove it at some point.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/ddp/participant/ParticipantDto.java [hunk 2]
```diff
@@ -101,6 +109,10 @@ public Builder(int ddpInstanceId, long lastChanged) {
             this.lastChanged = lastChanged;
         }
 
+        public Builder(int ddpInstanceId) {
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dto/ddp/participant/ParticipantDto.java:112]: All callers use the current time anyway, so at some point we can eliminate the other constructor.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantAdminOperationService.java [hunk 3]
```diff
@@ -41,7 +41,7 @@ protected boolean isBooleanProperty(String property, Map<String, String> attribu
 
     protected DDPInstance getDDPInstance(String realm, List<String> validRealms) {
         if (!validRealms.contains(realm.toLowerCase())) {
-            throw new DsmInternalError("Invalid realm for ParticipantDataFixupService: " + realm);
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/adminoperation/ParticipantAdminOperationService.java:44]: Typo.

File: pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/DDPMedicalRecordDataRequest.java [hunk 4]
```diff
@@ -120,14 +71,23 @@ private static List<Integer> writeInstitutionInfo(Connection conn, InstitutionRe
...
-                        institution, instanceName));
+                        institution, ddpInstance.getName()));
             }
         } else {
             log.info("Institution list was empty for participant {}", ddpParticipantId);
         }
         return medicalRecordIds;
     }
 
+    public static void updateMedicalRecordLog(Connection conn, String ddpParticipantId, int instanceId) {
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/DDPMedicalRecordDataRequest.java:82]: This is needed temporarily as MedicalRecord services are moved from here to `MedicalRecordService`.

File: pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/util/ElasticSearchParticipantExporterFactoryTest.java [hunk 5]
```diff
@@ -5,10 +5,12 @@
 import org.broadinstitute.dsm.util.export.NewOsteoParticipantExporter;
 import org.broadinstitute.dsm.util.export.ParticipantExportPayload;
 import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
 public class ElasticSearchParticipantExporterFactoryTest {
 
+    @Ignore
     @Test
```
Comment (denniscunningham) [inline pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/util/ElasticSearchParticipantExporterFactoryTest.java:14]: NewOsteoExporter is going away.

**Review by ssettipalli — APPROVED** [review]
looks ok to me.

**Pull request merged**